packages/server/lib/controllers/auth/postApiKey.ts

```diff
@@ -0,0 +1,220 @@
+            error: { code: 'invalid_body', errors: zodErrorToHTTP(val.error) }
+        });
+        return;
+    }
+
+    const queryStringVal = queryStringValidation.safeParse(req.query);
+    if (!queryStringVal.success) {
+        res.status(400).send({
+            error: { code: 'invalid_query_params', errors: zodErrorToHTTP(queryStringVal.error) }
+        });
+        return;
+    }
+
+    const paramsVal = paramsValidation.safeParse(req.params);
+    if (!paramsVal.success) {
+        res.status(400).send({
+            error: { code: 'invalid_uri_params', errors: zodErrorToHTTP(paramsVal.error) }
+        });
+        return;
+    }
+
+    const { account, environment } = res.locals;
+    const { apiKey }: PostPublicApiKeyAuthorization['Body'] = val.data;
+    const queryString: PostPublicApiKeyAuthorization['Querystring'] = queryStringVal.data;
+    const { providerConfigKey }: PostPublicApiKeyAuthorization['Params'] = paramsVal.data;
+    const connectionConfig = queryString.params ? getConnectionConfig(queryString.params) : {};
+    const connectionId = queryString.connection_id || connectionService.generateConnectionId();
+    const hmac = 'hmac' in queryString ? queryString.hmac : undefined;
+    const isConnectSession = res.locals['authType'] === 'connectSession';
+
+    let logCtx: LogContext | undefined;
+    try {
+        logCtx = await logContextGetter.create(
+            {
+                operation: { type: 'auth', action: 'create_connection' },
+                meta: { authType: 'apikey' },
+                expiresAt: defaultOperationExpiration.auth()
+            },
+            { account, environment }
+        );
+        void analytics.track(AnalyticsTypes.PRE_API_KEY_AUTH, account.id);
+
+        if (!isConnectSession) {
+            const checked = await hmacCheck({ environment, logCtx, providerConfigKey, connectionId, hmac, res });
+            if (!checked) {
+                return;
+            }
+        }
+
+        const config = await configService.getProviderConfig(providerConfigKey, environment.id);
+        if (!config) {
+            await logCtx.error('Unknown provider config');
+            await logCtx.failed();
+            res.status(404).send({ error: { code: 'unknown_provider_config' } });
+            return;
+        }
+
+        const provider = getProvider(config.provider);
+        if (!provider) {
+            await logCtx.error('Unknown provider');
+            await logCtx.failed();
+            res.status(404).send({ error: { code: 'unknown_provider_template' } });
+            return;
+        }
+
+        if (provider.auth_mode !== 'API_KEY') {
+            await logCtx.error('Provider does not support API key auth', { provider: config.provider });
+            await logCtx.failed();
+            res.status(400).send({ error: { code: 'invalid_auth_mode' } });
+            return;
+        }
+
+        if (!(await isIntegrationAllowed({ config, res, logCtx }))) {
+            return;
```

Comment:
Do we have a default response when there's no response set in our handler?

Reply:
the response is done in the method itself, honestly not ideal but handy for repeating code like this